File: P1A5 Creacion de Clases/Factura - main.cpp
#include <iostream>
using namespace std;

class Factura{
public:
    string numPiezas;
    string descripcion;
    int cantidad;
    double precio;
    
    Factura(){
        numPiezas = "";
        descripcion = "";
        cantidad = 0;
        precio = 0.0;
    }
    
    double obtenerMontoFactura(){
        return cantidad*precio;
    }
    
};


int main(){
    Factura ferre;
    cout << "Numero de piezas: "; cin >> ferre.numPiezas;
    cout << "Descripcion: "; cin >> ferre.descripcion;
    cout << "Cantidad: "; cin >> ferre.cantidad;
    cout << "Precio: "; cin >> ferre.precio;
    
    if (ferre.cantidad < 0) ferre.cantidad = 0;
    if (ferre.precio < 0 ) ferre.precio = 0.0;
    
    cout << "Factura: " << ferre.obtenerMontoFactura();
    
    return 0;
}

File: P1A5 Creacion de Clases/Factura - main.cs
using System;
class Factura{
 
    public string numPiezas;
    public string descripcion;
    public int cantidad;
    public double precio;
    
    public Factura(){
        numPiezas = "";
        descripcion = "";
        cantidad = 0;
        precio = 0.0;
    }
    
    public double obtenerMontoFactura(){
        return cantidad * precio;
    }
}

class FacturaTest{
    static void Main(string[] args){
        Factura ferre = new Factura();
        
        Console.WriteLine("Numero de piezas: ");
        ferre.numPiezas = Console.ReadLine();
        
        Console.WriteLine("Descripcion: ");
        ferre.descripcion = Console.ReadLine();
        
        Console.WriteLine("Cantidad: ");
        ferre.cantidad = Convert.ToInt32(Console.ReadLine());
        
        Console.WriteLine("Precio: ");
        ferre.precio = Convert.ToDouble(Console.ReadLine());
        
        
        if (ferre.cantidad < 0) ferre.cantidad = 0;
        if (ferre.precio < 0.0) ferre.precio = 0.0;
        
        Console.WriteLine("\nFactura: " + ferre.obtenerMontoFactura());
    }
}